### ISSUE TITLE
Can't modify indexes in an @OrderNonDet Collection

### ISSUE THREAD

**Issue by smillst** [t-rasmud/checker-framework#126]
Removing the element at a particular index in an `@OrderNonDet` collection should not be allowed, even if the index is deterministic.   This is because the element at that particular index is nondeterministic, so removing means that the collection no long contains a deterministic set of elements.

```
    void modifiers(
            @OrderNonDet List<@Det String> ondList,
            @Det List<@Det String> detList,
            @NonDet int nondetIndex) {
        ondList.remove(0); // error 
        ondList.set(0, "");  // error
}
```

Full Test case: checker/tests/determinism/IndexModifiers.java

**Comment by t-rasmud** [comment]
An `@OrderNonDet` collection means that the elements are deterministic but their iteration order is not. So, the above example looks correct to me i.e it shouldn't report any errors.

**Comment by mernst** [comment]
In an `@OrderNonDet` collection, the first element is some random element.  So, on different executions the remainder (everything but the first element) may differ.

**Comment by t-rasmud** [comment]
> In an `@OrderNonDet` collection, the first element is some random element. 

We do not allow random (`@NonDet`) elements to be added to `@OrderNonDet` collections. A collection with such random elements must be `@NonDet`.

**Comment by mernst** [comment]
Instead of "random", I should have said "arbitrary".

Suppose that there is an `@OrderNonDet` collection containing A, B, and C.
Removing the element at index 0 might remove A, might remove B, or might remove C.
These leave the collection in a different state (that is, with different elements).
Therefore, it must be forbidden to remove the element at a particular numeric index, in an `@OrderNonDet` collection.

**Comment by t-rasmud** [comment]
That makes sense. I read the statement `ondList.remove(0);` as removing the element `0` from the collection and I was clearly wrong.
This a specific case for `@OrderNonDet` lists.

**Comment by t-rasmud** [comment]
Fixed in baaf55bebed392b9f0b5e82293a9b2f50d926aeb
